### PR TITLE
fix: add pointer null check on ecall function

### PR DESCRIPTION
**Pull request by dingelish**
reported by @pedrotbtl , @w-woloszyn , and Professor Bill Roscoe. thanks for their contribution!

### CODE REVIEW

**Comment by dingelish** [comment]
@pedrotbtl @woloszyn please take a look. thanks!

File: samplecode/tls/tlsclient/enclave/src/lib.rs [hunk 1]
```diff
@@ -287,6 +287,10 @@ impl Sessions {
 
 #[no_mangle]
 pub extern "C" fn tls_client_new(fd: c_int, hostname: * const c_char, cert: * const c_char) -> usize {
+    if hostname.is_null() {
```
Comment (volcano0dr) [inline samplecode/tls/tlsclient/enclave/src/lib.rs]:
```
if hostname.is_null() || cert.is_null() {
    return 0xFFFF_FFFF_FFFF_FFFF;
}
```
Please add null pointer check for tls_server_new. thanks!.

```
pub extern "C" fn tls_server_new(fd: c_int, cert: * const c_char, key: * const c_char) -> usize 

if key.is_null() || cert.is_null() {
    return 0xFFFF_FFFF_FFFF_FFFF;
}
```

**Review by volcano0dr — COMMENTED** [review]
@dingelish Please add null pointer check for tls_server_new. thanks!

**Comment by pedrotbtl** [comment]
Hi @dingelish, it looks good to us! Can you just,please, correct Wojciech's handle: it is @w-woloszyn. Also, Professor Bill Roscoe is another contributor to our analysis, and to finding this issue as a consequence - he does not have a github handle, though. Just making sure everyone is acknowledged so we are all on the same page! :-D

Many thanks for keeping us in the loop and for the fix!

**Pull request merged**